allennlp/data/dataset.py

```diff
@@ -77,7 +77,8 @@ def get_padding_lengths(self) -> Dict[str, Dict[str, int]]:
 
     def as_arrays(self,
                   padding_lengths: Dict[str, Dict[str, int]] = None,
-                  verbose: bool = True) -> Dict[str, List[numpy.array]]:
+                  verbose: bool = True) -> Dict[str, List[Union[numpy.array,
```

Comment:
@joelgrus, is this be another place where generics would work?  I'm not sure if it'd be more work than it's worth.  Basically, each `Field` would return a `DataArray`, which is either a single array, or a dictionary of arrays, and then this type would be `Dict[str, DataArray]`.